src/adapter/src/coord/sequencer.rs

```diff
@@ -3073,6 +3076,29 @@ impl<S: Append + 'static> Coordinator<S> {
         Ok(ExecuteResponse::AlteredObject(ObjectType::Secret))
     }
 
+    async fn sequence_alter_source(
+        &mut self,
+        _session: &Session,
+        plan: AlterSourcePlan,
+    ) -> Result<ExecuteResponse, AdapterError> {
+        if let Some(host_config) = plan.config {
+            let host_config = self.catalog.resolve_storage_host_config(host_config)?;
+
+            let _source = match self.catalog.get_entry(&plan.id).item() {
+                CatalogItem::Source(source) => source,
+                other => coord_bail!("ALTER SOURCE entry was not a source: {other:?}"),
+            };
+            // TODO: persist the updated source definition
```

Comment:
This is the one bit that I definitely know I need to do, but am not sure how to manage yet.

Reply:
(I've now added a draft of a possible approach to this, but I'm not especially confident that it's the correct approach.)